evaluations/dotnix_2_piewol.md

```diff
@@ -7,15 +7,73 @@
-| **0d.** | Docker | <ul><li>[ ] </li></ul> | [docker.nix](https://github.com/sporyon/dotnix-core/blob/470096931e56374ebc3fcc0af821dd774ef44d10/docker.nix) | — |
-| 1. | Audit Trail | <ul><li>[ ] </li></ul> | [trail](https://github.com/sporyon/dotnix-core/blob/470096931e56374ebc3fcc0af821dd774ef44d10/pkgs/list-dependencies.nix) |  |
+| **0a.** | License | <ul><li>[x] </li></ul> | [LICENSE](https://github.com/sporyon/dotnix-core/blob/470096931e56374ebc3fcc0af821dd774ef44d10/LICENSE) | MIT |
+| **0b.** | Documentation | <ul><li>[x] </li></ul> | [README](https://github.com/sporyon/dotnix-core/blob/9a0cc3705e10da23fb2743b1a2f76fd8f2c4c3c4/README.md) | — |
+| **0c.** | Testing and Testing Guide | <ul><li>[ ] </li></ul> | [/checks](https://github.com/sporyon/dotnix-core/tree/470096931e56374ebc3fcc0af821dd774ef44d10/checks) | Testing guide looks fine but doesn't work atm. |
+| **0d.** | Docker | <ul><li>[ ] </li></ul> | [docker.nix](https://github.com/sporyon/dotnix-core/blob/470096931e56374ebc3fcc0af821dd774ef44d10/docker.nix) | following the instructions doesn't work |
+| 1. | Audit Trail | <ul><li>[ ] </li></ul> | [trail](https://github.com/sporyon/dotnix-core/blob/470096931e56374ebc3fcc0af821dd774ef44d10/pkgs/list-dependencies.nix) | doesn't work within docker |
 | 2 | CLI Tool Development | <ul><li>[ ] </li></ul> | [CLI](https://github.com/sporyon/dotnix-core/blob/470096931e56374ebc3fcc0af821dd774ef44d10/nixosModules/polkadot-validator.nix) | — |
 | 2a. | Functions| <ul><li>[ ] </li></ul> | [polkadot-validator.nix](https://github.com/sporyon/dotnix-core/blob/470096931e56374ebc3fcc0af821dd774ef44d10/nixosModules/polkadot-validator.nix) | — |
 
 
 ## General Notes
+sadly the validator doesn't seem to work when trying to run it in Docker.
+<details>
+
+```
+● polkadot-validator.service - Polkadot Validator
+     Loaded: loaded (/etc/systemd/system/polkadot-validator.service; linked; preset: enabled)
+     Active: activating (auto-restart) (Result: exit-code) since Tue 2024-07-30 22:52:36 UTC; 10s ago
+       Docs: https://github.com/paritytech/polkadot
+    Process: 514 ExecStart=/nix/store/88ikqbzkq5d7vnlj77l6gjbpw6ph23nx-polkadot-1.14.0/bin/polkadot --validator --name=dotnix-docker --chain=westend --base-pa>
+   Main PID: 514 (code=exited, status=1/FAILURE)
+         IP: 1.0K in, 780B out
+        CPU: 45ms
+```
+turns out that there seems to be an issue with setting the node key. Even after running
+```
+$ polkadot key generate-node-key | polkadot-validator --set-node-key
+```
+The shortened error output after running 
+```
+$ journalctl -n 1000 -f -u polkadot-validator.service
+```
+
+```Jul 30 23:10:27 dfc0bf67828a systemd[1]: Started Polkadot Validator.
+Jul 30 23:10:27 dfc0bf67828a polkadot[528]: Error:
+Jul 30 23:10:27 dfc0bf67828a polkadot[528]:    0: Starting an authorithy without network key in /run/credentials/polkadot-validator.service/node_key.
```

Comment:
Can you provide the exact commands you have entered after starting the docker image and the full journal? Entering just `polkadot key generate-node-key | polkadot-validator --set-node-key` should start the validator. This functionality has been implemented as part of the previous milestone and has not been touched since.

We could reproduce the issue by first generating the key with `polkadot key generate-node-key | polkadot-validator --set-node-key` which successfully starts the validator, and subsequently resetting the key to the empty string by running `polkadot-validator --set-node-key` and pressing enter, which will restart the validator into the broken state.

At the moment set node key will not check if the provided key is valid. This is something which needs improvement in the future.